homeassistant/components/nightscout/config_flow.py

```diff
@@ -38,21 +39,26 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         errors = {}
+
         if user_input is not None:
+            unique_id = hash_from_url(user_input[CONF_URL])
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
             try:
                 info = await _validate_input(user_input)
-            except CannotConnect:
-                errors["base"] = "cannot_connect"
-            except Exception:  # pylint: disable=broad-except
```

Comment:
Let's keep the unknown exception handling here. We usually have that in the config flow but not later.